src/metaswitch/clearwater/queue_manager/test/test_fvtests.py

```diff
@@ -0,0 +1,165 @@
+        self.assertFalse(self.plugins[0].at_front_of_queue_called)
+        self.assertFalse(self.plugins[1].at_front_of_queue_called)
+
+        # Add nodes to queue
+        self.add_all_nodes_to_queue("false")
+
+        # Check output
+        self.assertTrue(wait_for_success_or_fail(self.dummy_sync, partial(is_queue_config_as_expected, 0, 0)))
+        self.assertTrue(self.plugins[0].at_front_of_queue_called)
+        self.assertTrue(self.plugins[1].at_front_of_queue_called)
+
+    # Test that a twenty node queue is processed appropriately
+    @unittest.skipUnless(os.environ.get("SLOW"), "SLOW=T not set")
+    def test_twenty_node_queue(self):
+        self.create_synchronizers(20)
+
+        # Set up the plugins
+        for count in range(20):
+            self.plugins[count].set_front_of_queue_callback(partial(self.dummy_sync.remove_from_queue, True, self.syncs[count]._id))
+            self.assertFalse(self.plugins[count].at_front_of_queue_called)
+
+        # Add nodes to queue
+        self.add_all_nodes_to_queue("false")
+
+        # Check output
+        self.assertTrue(wait_for_success_or_fail(self.dummy_sync, partial(is_queue_config_as_expected, 0 , 0)))
+        for plugin in self.plugins:
+            self.assertTrue(plugin.at_front_of_queue_called)
+
+    # Test that a twenty node queue where every node is marked as failed is
+    # processed appropriately
+    @unittest.skipUnless(os.environ.get("SLOW"), "SLOW=T not set")
+    def test_twenty_node_queue_force_true(self):
+        self.create_synchronizers(20)
+
+        # Set up the plugins
+        for count in range(20):
+            self.plugins[count].set_front_of_queue_callback(partial(self.dummy_sync.remove_from_queue, False, self.syncs[count]._id))
+            self.assertFalse(self.plugins[count].at_front_of_queue_called)
+
+        # Add nodes to queue
+        self.add_all_nodes_to_queue("true")
+
+        # Check output
+        self.assertTrue(wait_for_success_or_fail(self.dummy_sync, partial(is_queue_config_as_expected, 20, 0)))
+        for plugin in self.plugins:
+            self.assertTrue(plugin.at_front_of_queue_called)
+
+    # Test that a two node queue where every node fails aborts after the first failure if
+    # force is false
+    @unittest.skipUnless(os.environ.get("SLOW"), "SLOW=T not set")
+    def test_two_node_queue_force_false(self):
+        self.create_synchronizers(2)
+
+        # Set up the plugins
+        for count in range(2):
+            self.plugins[count].set_front_of_queue_callback(partial(self.dummy_sync.remove_from_queue, False, self.syncs[count]._id))
+            self.assertFalse(self.plugins[count].at_front_of_queue_called)
+
+        # Add nodes to queue
+        self.add_all_nodes_to_queue("false")
+
+        # Check output
+        self.assertTrue(wait_for_success_or_fail(self.dummy_sync, partial(is_queue_config_as_expected, 1, 0)))
```

Comment:
I'm not really sure what real-world condition we're asserting on here - comment?